sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -436,6 +429,49 @@ protected ServiceBusProcessor()
             }
         }
 
+        /// <summary>
+        /// Invokes the process message event handler after a message has been received.
+        /// This method can be overriden to raise an event manually for testing purposes.
+        /// </summary>
+        /// <param name="args">The event args containing information related to the message.</param>
+        protected internal virtual async Task OnProcessMessageAsync(ProcessMessageEventArgs args)
+        {
+            await _processMessageAsync(args).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Invokes the error event handler when an error has occured during processing.
+        /// This method can be overriden to raise an event manually for testing purposes.
+        /// </summary>
+        /// <param name="args">The event args containing information related to the error.</param>
+        protected internal async virtual Task OnProcessErrorAsync(ProcessErrorEventArgs args)
+        {
+            try
+            {
+                await _processErrorAsync(args).ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                // don't bubble up exceptions raised from customer exception handler
```

Comment:
This scares me, as callers may not understand that exceptions are being swallowed.  This may be the pattern that our current code uses, but may not be clear to future maintainers nor to developers extending this type and delegating to the `protected` member.

Have you considered throwing here and explicitly swallowing at the call site to make the decision clear at a glance?   If you decide to keep this form, can we please call out the behavior in the `<remarks>`?